sys/netbsd/sys.txt

```diff
@@ -113,6 +114,9 @@ wait4(pid pid, status ptr[out, int32, opt], options flags[wait_options], ru ptr[
 exit(code intptr)
 ptrace(request intptr, pid pid, addr ptr[out, intptr], data intptr)
 __clone(flags flags[clone_flags], sp buffer[in]) pid
+fork() pid
+vfork() pid
+posix_spawn(pid pid, path ptr[in, string], file_actions ptr[in, posix_spawn_file_actions], attrp ptr[in, posix_spawnattr], argv ptr[in, array[string]], envp ptr[in, array[string]]) pid
```

Comment:
It seems there is something missing that would allow kernel to infer length of argv/envp arrays.

Reply:
The arguments are correct, I verified. And looks similar on Linux also.

Reply:
The raw types may be correct, but I suspect we are missing some bit of _semantic_ information. E.g. an empty string at the end?
We try to describe as much semantic information as possible because it makes guessing right combinations much easier. Otherwise, all arguments of all syscalls may be described as intptr and formally it is a correct type ;)

Reply:
The argument type for envp and argv is `char *const *`
My representation of that is correct right?

Reply:
No, not correct.
The rule is simple: if there are two `*` the syzkaller type must contain two `ptr` types.
So that would be `ptr[in, array[ptr[in, string]]]`.
What you described is a pointer to a single block of memory that contains multiple strings one after another.

Reply:
But a string itself would be a pointer right(we would be storing the string at some point and passing a pointer to it)? So there are already two pointers, right?

Reply:
No! Looks at the second path argument. String is literally that -- array of bytes, or array[int8].

Reply:
Ah. Makes sense. Sorry for the confusion.